03-rnaseq/pathway-analysis_rnaseq_03_gsva.Rmd

```diff
@@ -435,12 +435,222 @@ sum(duplicated(mapped_df$entrez_id))
+    "SRP140558_gsva_results.tsv"
+  ))
+```
+
+## Visualizing results with a heatmap
+
+Let's make a heatmap for our pathways! 
+
+### Neaten up our metadata labels
+
+We will want our heatmap to include our metadata labels, but unfortunately some of the metadata for this dataset are not set up into separate, neat columns. 
+
+Our main information about these samples is squished into one column, `refinebio_title`. 
+Let's preview what this column looks like. 
+
+```{r}
+head(metadata$refinebio_title)
+```
+
+This information is not very clear as is. 
+
+Looking at the author's descriptions, PBMCs were collected at two time points: during the patients' first, acute bronchiolitis visit (abbreviated "AV") and their recovery visit, (called post-convalescence and abbreviated "CV").
+
+We can create a new variable, `time_point`, that states this info more clearly. 
+This new `time_point` variable will have two labels: `acute illness` and `recovering` based on the `AV` or `CV` coding located in the `refinebio_title` string variable. 
+
+```{r}
+annot_df <- metadata %>%
+  # We need the sample IDs and the main column that contains the metadata info
+  dplyr::select(
+    refinebio_accession_code,
+    refinebio_title
+  ) %>%
+  # Create our `time_point` variable based on `refinebio_title`
+  dplyr::mutate(
+    time_point = dplyr::case_when(
+      # Create our new variable based on refinebio_title containing AV/CV
+      stringr::str_detect(refinebio_title, "_AV_") ~ "acute illness",
+      stringr::str_detect(refinebio_title, "_CV_") ~ "recovering"
+    )
+  ) %>%
+  # We don't need the older version of the variable anymore
+  dplyr::select(-refinebio_title)
+```
+
+These time point samples are paired, so you could also add the `refinebio_subject` to the labels, but for simplicity we've left them off for now. 
+
+The `pheatmap::pheatmap()` will want the annotation data frame to have matching row names to the data we supply it (which is our `gsva_results`).
+
+```{r}
+annot_df <- annot_df %>%
+  # pheatmap will want our sample names that match our data to
+  tibble::column_to_rownames("refinebio_accession_code")
+```
+
+### Set up the heatmap itself
+
+Great! We're all set. 
+We can see that they are in a wide format with the GSVA scores for each sample spread across a row associated with each pathway.
+
+```{r}
+pathway_heatmap <- pheatmap::pheatmap(gsva_results,
+  annotation_col = annot_df, # Add metadata labels!
+  show_colnames = FALSE, # Don't show sample labels
+  fontsize_row = 6 # Shrink the pathway labels a tad
+)
+
+# Print out heatmap here
+pathway_heatmap
+```
+
+Here we've used clustering and can see that samples somewhat cluster by `time_point`. 
+
+We can also see the conceptually similar pathways seem to cluster together (e.g. `HALLMARK_INTERFERON_ALPHA_RESPONSE` and `HALLMARK_INTERFERON_GAMMA_RESPONSE`). 
+Often pathways may look similar because their gene sets are not distinct from each other and can have a lot of overlap. 
```

Comment:
I tried to just stick to the basic idea here that I was trying to get across but added explanations to it.